? compare_views_block_hash.patch
Index: views.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/views/views.module,v
retrieving revision 1.341.2.4
diff -u -r1.341.2.4 views.module
--- views.module	5 Jan 2010 00:04:30 -0000	1.341.2.4
+++ views.module	14 Jan 2010 15:40:24 -0000
@@ -349,7 +349,11 @@
         }
       }
 
-      variable_set('views_block_hashes', $hashes);
+      // Only save hashes if they have changed.
+      $old_hashes = variable_get('views_block_hashes', array());
+      if ($hashes != $old_hashes) {
+        variable_set('views_block_hashes', $hashes);
+      }
       // Save memory: Destroy those views.
       foreach ($views as $view) {
         $view->destroy();
